File: 프로그래머스/Level1/[1차] 비밀지도.cpp
#include <string>
#include <vector>

using namespace std;
vector<string> sharp;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    
    for(int i=0; i<n; i++){
        answer.push_back("");
        
        for(int j=0; j<n; j++){
            
            if(arr1[i]%2 || arr2[i]%2) sharp.push_back("#");
            else sharp.push_back(" ");
            
            arr1[i]=arr1[i]/2;
            arr2[i]=arr2[i]/2;
        }
        
        for(int k=n-1; k>=0; k--){
            answer[i] += sharp[k];
        }
        sharp.clear();
    }
    

    return answer;
}

File: 프로그래머스/Level1/[1차] 비밀지도.js
function solution(n, arr1, arr2) {
    let tmp1 = arr1.map(elem => elem.toString(2));
    let tmp2 = arr2.map(elem => elem.toString(2));
    let answer = [];
    
    for(let i=0; i<n; i++){
        while(tmp1[i].length !== n)
            tmp1[i] = '0' + tmp1[i];
        
        while(tmp2[i].length !== n)
            tmp2[i] = '0' + tmp2[i];
    }
    
    for(let i=0; i<n; i++){
        answer.push('');
        
        for(let j=0; j<n; j++){
            if(tmp1[i][j] === '1' || tmp2[i][j] === '1')
                answer[i] += '#';
            else answer[i] += ' ';
        }
    }
    
    return answer;
}
